File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int i = 0, ans = 0, n = s.length();
        for(int j = 0; j < n; ++j) {
            maxCost -= abs(s[j]-t[j]);
            while(maxCost < 0 && i <= j && i < n) {
                maxCost += abs(s[i]-t[i]);
                ++i;
            }
            ans = max(ans, j-i+1);
        }
        return ans;
    }
};

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.cs
public class Solution {
    public int EqualSubstring(string s, string t, int maxCost) {
        int i = 0, ans = 0, n = s.Length;
        for (int j = 0; j < n; ++j)
        {
            maxCost -= Math.Abs(s[j] - t[j]);
            while (maxCost < 0 && i <= j && i < n)
            {
                maxCost += Math.Abs(s[i] - t[i]);
                ++i;
            }

            ans = Math.Max(ans, j - i + 1);
        }

        return ans;
    }
}